drivers/pq.go

```diff
@@ -13,9 +13,14 @@ type Pq struct {
 	stmt *sql.Stmt
 }
 
+func (d *Pq) Clone() Driver {
```

Comment:
I don't understand how this is making a copy. It appears to just be dereferencing the pointer, then returning the address of that same object.

Reply:
This explains it. I'm not sure go makes all this stuff simpler. This is going to be an interview question I run into at some point in the future. I just know it.

https://stackoverflow.com/questions/38443348/does-dereferencing-a-struct-return-a-new-copy-of-struct

Reply:
huh, ok.